haralyzer/assets.py

```diff
@@ -324,7 +332,16 @@ def filter_entries(self, request_type=None, content_type=None,
             if http_version is not None and not p.match_http_version(
                     entry, http_version, regex=regex):
                 valid_entry = False
-
+            if source is not None and source == 'internal':
+                if p.hostname == get_entry_hostname(entry):
+                    valid_entry = True
```

Comment:
Good suggestion. I did notice that while writing the changes, however, I just wanted to get it working before making it prettier. I'll get that updated ASAP.